pages/home.jsx

```diff
@@ -5,15 +5,156 @@ var Link = Router.Link;
+          </div>
+        </div>
+      </div>
+    );
+  }
+});
+
+var FeaturedPost = React.createClass({
+  propTypes: {
+    data: React.PropTypes.object.isRequired
+  },
+  render: function() {
+    var publishedDate = new Date(this.props.data.publishedDate);
+    var month = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"][publishedDate.getMonth()];
+    return(
+      <div id="featured-post">
+        <div className="entry-posted-container">
+          <p className="entry-posted">
+            <time className="published" title={this.props.data.publishedDate} dateTime={this.props.data.publishedDate} >
+              <span className="posted-month">{month}</span>
+              <span className="posted-date">{publishedDate.getDate()}</span>
+              <span className="posted-year">{publishedDate.getFullYear()}</span>
+            </time>
+          </p>
+        </div>
+        <div className="entry-header-container">
+          <h3 className="entry-title"><a href={this.props.data.link}>{this.props.data.title}</a></h3>
+          <cite className="author">{this.props.data.author}</cite>
+        </div>
+        <p className="excerpt">
+          {this.props.data.contentSnippet}
+        </p>
+        <a className="more" href={this.props.data.link}>Continue reading</a>
+      </div>
+    );
+  },
+});
+
+var LatestPosts = React.createClass({
+  propTypes: {
+    data: React.PropTypes.array.isRequired
+  },
+  render: function() {
+    function formatDate(theDate) {
+      return ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"][theDate.getMonth()] + " " +
+        theDate.getDate() + ", " +
+        theDate.getFullYear();
+    }
+    return (
+      <ul id="recent-posts">
+        {
+          this.props.data.map(function(post, i) {
+            return (
+              <li key={i}>
+                <a className="post-title">{post.title}</a>
+                <time className="published" title={post.publishedDate} dateTime={post.publishedDate}>
+                  <span>{formatDate(new Date(post.publishedDate))}</span>
+                </time>
+              </li>
+            )
+          })
+        }
+      </ul>
+    );
+  }
+});
+
+var BlogSection = React.createClass({
+  getInitialState: function() {
+    return {
+      featuredPostData: {
+        title: "",
+        author: "",
+        publishedDate: "",
+        contentSnippet: "",
+        link: ""
+      },
+      latestPostsData: []
+    }
+  },
+  loadGoogleAPI: function(callback) {
+    var head = document.getElementsByTagName("head")[0];
+    var gScript = document.createElement("script");
+    gScript.setAttribute("src", "https://www.google.com/jsapi");
+    gScript.onload = callback;
+    head.appendChild(gScript);
+    this.setState({googleAPILoaded: true});
+  },
+  componentDidMount: function() {
+    var self = this;
+    var google;
+    this.loadGoogleAPI(function() {
+      google = window.google;
```

Comment:
Might want to guard these callbacks with a check to see if [`this.isMounted()`](https://facebook.github.io/react/docs/component-api.html#ismounted) is true. Otherwise, for instance, if the network is slow, the user may navigate to a different section of the site before the callback is called, and then an exception will be thrown (and unnecessary network activity may occur) when the callback is finally triggered.